graspy/inference/latent_distribution_test.py

```diff
@@ -287,17 +307,74 @@ def fit(self, A1, A2):
 
         Parameters
         ----------
-        A1, A2 : nx.Graph, nx.DiGraph, nx.MultiDiGraph, nx.MultiGraph, np.ndarray
-            The two graphs to run a hypothesis test on.
+        A1, A2 : variable (see description)
+            The two graphs, or their embeddings to run a hypothesis test on.
+            Expected variable type and shape depends on input_graph attribute:
+
+            - input_graph=True
+                expects two unembedded graphs either as NetworkX graph objects, or as
+                two np.ndarrays, representing the adjacency matrices. In this
+                case will be embedded using adjacency spectral embedding.
+            - input_graph-False
+                expects two already embedded graphs. In this case they must be
+                arrays of shape (n, d) and (m, d), where d, the number of
+                components, must be shared.
+
+            Note that regardless of how the graphs are passed, they need not
+            have the same number of vertices.
 
         Returns
         -------
         self
         """
-        A1 = import_graph(A1)
-        A2 = import_graph(A2)
+        if self.input_graph:
+            A1 = import_graph(A1)
+            A2 = import_graph(A2)
 
-        X1_hat, X2_hat = self._embed(A1, A2)
+            X1_hat, X2_hat = self._embed(A1, A2)
+        else:
+            # check for nx objects, since they are castable to arrays,
+            # but we don't want that
+            if isinstance(A1, nx.Graph) or isinstance(A1, nx.DiGraph):
+                msg = (
+                    f"embedding of the first graph is an {type(A1)}, not an array! "
+                    "if input_graph is False, the inputs need to be adjacency "
+                    "spectral embeddings, with shapes (n, d) and (m, d)"
+                )
+                raise TypeError(msg)
+            if isinstance(A2, nx.Graph) or isinstance(A2, nx.DiGraph):
+                msg = (
+                    f"embedding of the second graph is an {type(A2)}, not an array! "
+                    "if input_graph is False, the inputs need to be adjacency "
+                    "spectral embeddings, with shapes (n, d) and (m, d)"
+                )
+                raise TypeError(msg)
+
+            # allowing d!=2, and ndim>2 in order to throw custom errors below
+            X1_hat = check_array(A1, ensure_2d=False, allow_nd=True)
```

Comment:
apologies for being unclear about this earlier and making things more confusing. I think we need: 
1. check that A1, A2 are np.ndarray
2. check that they are kosher arrays
sklearn `check_array` does 2 only. Right now if you pass in a float or something silly, the error message doesn't make a ton of sense. Can you just "invert" the `isinstance(A1, nx.Graph)` to `if not isinstance(A1, np.ndarray)` and throw a descriptive error? I think it is worth saying here "if input_graph is true... then expects blah blah blah" cause this will be an easy mistake to make if you aren't paying attention to `input_graph` parameter

Reply:
I don't have an issue with using sklearn `check_array` to ensure 2d input. if you would rather be explicit i am ok with it.

Reply:
so this is why i was asking whether there is a graceful way of allowing arrays, sparse arrays and etc., and j1 recommended to use this. as long as i explicitly prohibit nx objects which geet casted and throw a 1-d error, while it should be a "not an array error", it rejects all improper inputs, but the error message could certainly be better.

i thought about it a little more, and i think that i've overcomplicated by trying to allow sparse arrays, and etc. these are embeddings, and they really should be ndarrays and nothing else. i am going to just change to checking if it is a numpy array.

i prefer my descriptive error to check_array error. it's not as bad as their object type error, but i still want to be explicit to the user.

okay, so in this case - check array isn't even doing much (normally it casts, checks for ndim, 2d-ness, etc. but now we're doing all of it ourselves), so i'll remove it altoghether.

Reply:
okay, will add back in the check_array for the inf stuff, and also in case they are checking for some other kosher-ity that i am not aware off.